runtime/lua/vim/loader.lua

```diff
@@ -116,6 +116,13 @@ end
 --- @param name string can be a module name, or a file name
 --- @return string file_name
 local function cache_filename(name)
+  -- Simplify the name of AppImage mounted file, to prevent
+  -- random part in it from creating cache misses.
+  if os.getenv('APPIMAGE') ~= nil then
+    local idx = name:find('/usr/', 1, true) --- @type integer?
```

Comment:
look at examples in the codebase, if in doubt
```suggestion
    local idx = name:find('/usr/', 1, true) ---@type integer?
```